lib/index.js

```diff
@@ -53,13 +53,40 @@ function applyMethods(q) {
     });
   };
 
-  q.series = function(arr, body) {
+  q.series = function (arr, body) {
     return arr.reduce(function(promise, item) {
       return promise.then(function(result) {
         return body(item, result);
       });
     }, q());
   };
+
+  /**
+   * Allows body up to timeout ms to complete, and retries it if exceeds that time.
+   * Body will not be retried if it fails for any other reason (e.g. exception or rejection).
+   */
+  q.timeoutRetry = function (timeout, body, options) {
+    options = options || {};
+
+    var timeoutIdentifier = 'timeout-dc1836808a';
+    var maxAttempts = options.maxAttempts || 5;
+    var attempts = 0;
+    var success = false;
+
+    return q.while(function () {
+      return !success && attempts < maxAttempts;
+    }, function () {
+      attempts += 1;
```

Comment:
Perhaps it's overkill, for now, but I do think exponential backoff and perhaps even jitter do belong here.
You could just adjust the timeout, based on the number of attempts so far.